Tasks/Npm/npmtask.ts

```diff
@@ -13,15 +16,85 @@ var command = tl.getInput('command', true);
 if (command.indexOf(' ') >= 0) {
 	tl.setResult(tl.TaskResult.Failed, tl.loc("InvalidCommand"));
 }
-npm.arg(command);
-
-npm.argString(tl.getInput('arguments', false));
-
-npm.exec()
-.then(function(code) {
-	tl.setResult(code, tl.loc('NpmReturnCode', code));
-})
-.fail(function(err) {
-	tl.debug('taskRunner fail');
-	tl.setResult(tl.TaskResult.Failed, tl.loc('NpmFailed', err.message));
-})
\ No newline at end of file
+
+if(tl.getVariable('USE_DEPRECATED_TASK_VERSION') != undefined && tl.getVariable('USE_DEPRECATED_TASK_VERSION').toLowerCase() === 'true') {
```

Comment:
nit: space after "if"
Move "USE_..._VERSION" to a constant?
Could this entire condition be refactored into a method that is more descriptive?  shouldUseDeprecatedTask()